cirq-core/cirq/devices/noise_utils.py

```diff
@@ -78,20 +79,21 @@ def __str__(self):
         return f'{self.gate_type}{self.qubits}'
 
     def __repr__(self) -> str:
-        fullname = f'{self.gate_type.__module__}.{self.gate_type.__qualname__}'
         qubits = ', '.join(map(repr, self.qubits))
-        return f'cirq.devices.noise_utils.OpIdentifier({fullname}, {qubits})'
+        return f'cirq.devices.noise_utils.OpIdentifier({proper_repr(self.gate_type)}, {qubits})'
 
     def _value_equality_values_(self) -> Any:
         return (self.gate_type, self.qubits)
 
     def _json_dict_(self) -> Dict[str, Any]:
-        gate_json = protocols.json_cirq_type(self._gate_type)
-        return {'gate_type': gate_json, 'qubits': self._qubits}
+        if hasattr(self.gate_type, '__name__'):
+            return {'gate_type': protocols.json_cirq_type(self.gate_type), 'qubits': self._qubits}
```

Comment:
Use internal vars in the classes.
```suggestion
            return {'gate_type': protocols.json_cirq_type(self._gate_type), 'qubits': self._qubits}
```

Reply:
Done.